sensorprocess/sensorprocess.go

```diff
@@ -12,39 +11,85 @@ import (
 	s "github.com/viamrobotics/viam-cartographer/sensors"
 )
 
-// defaultTime is used to check if timestamps have not been set yet.
-var defaultTime = time.Time{}
-
 // Config holds config needed throughout the process of adding a sensor reading to the cartofacade.
 type Config struct {
-	CartoFacade            cartofacade.Interface
-	sensorProcessStartTime time.Time
-	IsOnline               bool
-
-	Lidar            s.TimedLidar
-	currentLidarData *s.TimedLidarReadingResponse
+	CartoFacade cartofacade.Interface
+	IsOnline    bool
 
-	IMU            s.TimedMovementSensor
-	currentIMUData *s.TimedIMUReadingResponse
+	Lidar s.TimedLidar
+	IMU   s.TimedMovementSensor
 
 	Timeout                  time.Duration
 	InternalTimeout          time.Duration
 	Logger                   logging.Logger
 	RunFinalOptimizationFunc func(context.Context, time.Duration) error
-
-	Mutex *sync.Mutex
 }
 
-// Update currentIMUData under a mutex lock.
-func (config *Config) updateMutexProtectedIMUData(data s.TimedIMUReadingResponse) {
-	config.Mutex.Lock()
-	config.currentIMUData = &data
-	config.Mutex.Unlock()
+// StartOfflineSensorProcess starts the process of adding lidar and movement sensor data
+// in a deterministically defined order to cartographer.
+func (config *Config) StartOfflineSensorProcess(ctx context.Context) bool {
+	// get the initial lidar reading
+	lidarReading, lidarEndOfDataSetReached, err := getTimedLidarReading(ctx, config)
+	if err != nil || lidarEndOfDataSetReached {
```

Comment:
[q] why if err != nil and lidarEndOfDataSetReached  is false do we return lidarEndOfDataSetReached?

Reply:
Ignoring `lidarEndOfDatasetReached`, the err can be `errEndOfDataset`. It could also potentially be another type of error. Why are we handling both the same?

Reply:
Will change this.